plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -333,6 +469,46 @@ private Boolean isModelNotDeployed(MLModelState mlModelState) {
             && !mlModelState.equals(MLModelState.PARTIALLY_DEPLOYED);
     }
 
+    private <T> Boolean isPipelineContainsModel(
+        List<T> pipelineConfigurations,
+        String candidateModelId,
+        Function<T, Map<String, Object>> getConfigFunction
```

Comment:
Is there a simple way to extract the configuration map from the responses, then change this method to accept Map<String, Object> as parameter?

Reply:
If we want to extract them before, we may need `instanceof` to check the type of input pipelineConfigurations.